lib/src/components/supa_email_auth.dart

```diff
@@ -120,98 +127,156 @@ class _SupaEmailAuthState extends State<SupaEmailAuth> {
-                        decoration: InputDecoration(
-                          label: Text(metadataField.label),
-                          prefixIcon: metadataField.prefixIcon,
+            spacer(16),
+            if (widget.metadataFields != null && !_isSigningIn)
+              ...widget.metadataFields!
+                  .map((metadataField) => [
+                        TextFormField(
+                          controller: _metadataControllers[metadataField],
+                          decoration: InputDecoration(
+                            label: Text(metadataField.label),
+                            prefixIcon: metadataField.prefixIcon,
+                          ),
+                          validator: metadataField.validator,
                         ),
-                        validator: metadataField.validator,
+                        spacer(16),
+                      ])
+                  .expand((element) => element),
+            ElevatedButton(
+              child: (_isLoading)
+                  ? SizedBox(
+                      height: 16,
+                      width: 16,
+                      child: CircularProgressIndicator(
+                        color: Theme.of(context).colorScheme.onPrimary,
+                        strokeWidth: 1.5,
                       ),
-                      spacer(16),
-                    ])
-                .expand((element) => element),
-          ElevatedButton(
-            child: (_isLoading)
-                ? SizedBox(
-                    height: 16,
-                    width: 16,
-                    child: CircularProgressIndicator(
-                      color: Theme.of(context).colorScheme.onPrimary,
-                      strokeWidth: 1.5,
-                    ),
-                  )
-                : Text(isSigningIn ? 'Sign In' : 'Sign Up'),
-            onPressed: () async {
-              if (!_formKey.currentState!.validate()) {
-                return;
-              }
-              setState(() {
-                _isLoading = true;
-              });
-              try {
-                late final AuthResponse response;
-                if (isSigningIn) {
-                  response = await supaClient.auth.signInWithPassword(
-                    email: _email.text,
-                    password: _password.text,
-                  );
-                } else {
-                  response = await supaClient.auth.signUp(
-                    email: _email.text,
-                    password: _password.text,
-                    emailRedirectTo: widget.redirectUrl,
-                    data: widget.metadataFields == null
-                        ? null
-                        : _metadataControllers.map<String, dynamic>(
-                            (metaDataField, controller) =>
-                                MapEntry(metaDataField.key, controller.text)),
-                  );
+                    )
+                  : Text(_isSigningIn ? 'Sign In' : 'Sign Up'),
```

Comment:
not related to this PR but would be nice to be able to modify the text that shows up. This would be useful to handle intl

Reply:
Yup, for sure! Certainly something we would want to do in the near future!